esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -5033,13 +5033,18 @@ bool CWsDfuEx::onListHistory(IEspContext &context, IEspListHistoryRequest &req,
         {
             IPropertyTree *history = file->queryHistory();
             if (history)
+            {
                 storeHistoryTreeToArray(history, arrHistory);
+                history->serialize(xmlmap);
```

Comment:
Sorry, I should have explained further.  Just setting the field is fine for anything that is not too expensive.  Versioning will remove it automatically if needed.  But for something large or expensive you can check the version and only set it if needed.

`if (context.getClientVersion() < 1.36)`
`   history->serialze(xmlmap)`;
`if (xmlmap.length())`
`   resp.setXmlmap(xmlmap);`